rst_files/wald_friedman.rst

```diff
@@ -401,467 +345,201 @@ that appear on the right side of the Bellman equation as separate functions
-Let's code this problem up and solve it
-
-To approximate the value function that solves Bellman equation :eq:`new1`, we
-use value function iteration
-
-* For earlier examples of this technique see the :doc:`shortest path <short_path>`, :doc:`job search <mccall_model>` or :doc:`optimal growth <optgrowth>` lectures
-
-As in the :doc:`optimal growth lecture <optgrowth>`, to approximate a continuous value function
-
-* We iterate at a finite grid of possible values of :math:`p`
-
-* When we evaluate :math:`A(p)` between grid points, we use linear interpolation
-
-This means that to evaluate :math:`J(p)` where :math:`p` is not a grid point, we must use two points:
-
-* First, we use the largest of all the grid points smaller than :math:`p`, and call it :math:`p_i`
-
-* Second, we use the grid point immediately after :math:`p`, named :math:`p_{i+1}`, to approximate the function value as
+Let's code this problem up and solve it.
 
-.. math::
-
-    J(p) = J(p_i) + (p - p_i) \frac{J(p_{i+1}) - J(p_i)}{p_{i+1} - p_{i}}
+We implement the cost functions for each choice considered in the
+Bellman equation :eq:`new3`. First consider the cost associated to accepting either
+distribution and compare the minimum of the two to the expected benefit of
+drawing again. Drawing again will only be worthwhile if the expected marginal
+benefit of learning from an additional draw is greater than the explicit cost.
 
-
-In one dimension, you can think of this as simply drawing a line between each pair of points on the grid
+For every belief :math:`p`, we can compute the difference between accepting a
+distribution and choosing to draw again. The solution :math:`\alpha` and :math:`\beta`
+occurs when the difference between the cost of drawing again and accepting a
+distribution are zero (i.e., the agent is indifferent). We can use any roots
+finding algorithm to solve for the solutions in the interval [0, 1]. Lastly,
+verify which indifference points correspond to the definition.
 
 Here's the code
 
 .. code-block:: julia
 
-    expect_loss_choose_0(p, L0) = (1 - p) * L0
-
-    expect_loss_choose_1(p, L1) = p * L1
-
-    function EJ(p, f0, f1, J)
-        # Get the current distribution we believe (p * f0 + (1 - p) * f1)
-        curr_dist = p * f0 + (1 - p) * f1
-
-        # Get tomorrow's expected distribution through Bayes law
-        tp1_dist = clamp.((p * f0) ./ (p * f0 + (1 - p) * f1), 0, 1)
-
-        # Evaluate the expectation
-        EJ = dot(curr_dist, J.(tp1_dist))
-
-        return EJ
+    accept_x0(p, L0) = (one(p) - p) * L0
+    accept_x1(p, L1) = p * L1
+    bayes_update(p, d0, d1) = p * pdf(d0, p) / pdf(MixtureModel([d0, d1], [p, one(p) - p]), p)
+    function draw_again(p, d0, d1, L0, L1, c, target)
+        candidate = 0.0
+        cost = 0.0
+        while candidate < target
+            p = bayes_update(p, d0, d1)
+            cost += c
+            candidate = min(accept_x0(p, L0), accept_x1(p, L1)) + cost
+            candidate ≥ target && break
```

Comment:
https://github.com/QuantEcon/lecture-source-jl/blob/master/style.md#naming-conventions-comments-etc  ascii for control flow